clients/python-wrapper/lakefs/client.py

```diff
@@ -106,6 +112,125 @@ def version(self) -> str:
+   :raises ValueError: If the session does not have a region name set.
+   """
+
+    # this method should only be called when installing the aws-iam additional requirement
+    from botocore.client import Config  # pylint: disable=import-outside-toplevel, import-error
+    from botocore.signers import RequestSigner  # pylint: disable=import-outside-toplevel, import-error
+
+    sts_client = session.client('sts', config=Config(signature_version='v4'))
+    endpoint = sts_client.meta.endpoint_url
+    service_id = sts_client.meta.service_model.service_id
+    region = _extract_region_from_endpoint(endpoint)
+    # signer is used because the presigned URL generated by the STS does not support additional headers
+    signer = RequestSigner(
+        service_id,
+        region,
+        'sts',
+        'v4',
+        session.get_credentials(),
+        session.events
+    )
+    endpoint_with_params = f"{endpoint}/?Action=GetCallerIdentity&Version=2011-06-15"
+    if additional_headers is None:
+        additional_headers = {
+            'X-LakeFS-Server-ID': lakefs_host,
+        }
+    params = {
+        'method': 'POST',
+        'url': endpoint_with_params,
+        'body': {},
+        'headers': additional_headers,
+        'context': {}
+    }
+
+    presigned_url = signer.generate_presigned_url(
+        params,
+        region_name=region,
+        expires_in=presign_expression,
+        operation_name=''
+    )
+    parsed_url = urlparse(presigned_url)
+    query_params = parse_qs(parsed_url.query)
+
+    # Extract values from query parameters
+    print(query_params)
+    json_object = {
+        "method": "POST",
+        "host": parsed_url.hostname,
+        "region": region,
+        "action": query_params['Action'][0],
+        "date": query_params['X-Amz-Date'][0],
+        "expiration_duration": query_params['X-Amz-Expires'][0],
+        "access_key_id": query_params['X-Amz-Credential'][0].split('/')[0],
+        "signature": query_params['X-Amz-Signature'][0],
+        "signed_headers": query_params.get('X-Amz-SignedHeaders', [''])[0].split(';'),
+        "version": query_params['Version'][0],
+        "algorithm": query_params['X-Amz-Algorithm'][0],
+        "security_token": query_params.get('X-Amz-Security-Token', [None])[0]
+    }
+
+    json_string = json.dumps(json_object)
+    return base64.b64encode(json_string.encode('utf-8')).decode('utf-8')
+
+
+def from_aws_role(session: 'boto3.Session', ttl_seconds: int = 3600, **kwargs) -> Client:
+    """
+    Create a lakeFS client from an AWS role.
+    :param session: : The boto3 session.
+    :param ttl_seconds: The time-to-live for the generated token in seconds. The default value is 3600 seconds (1 hour).
+    :param kwargs: The arguments to pass to the client.
+    :return: A lakeFS client.
+    """
+
+    client = Client(**kwargs)
+    lakefs_host = urlparse(client.config.host).hostname
+    identity_token = _get_identity_token(session, lakefs_host)
```

Comment:
When calling `_get_identity_token` missing `additional_headers` and `presign_expression`.